README.md

```diff
@@ -9,3 +9,664 @@ ExpressRouterAdapter is available as a package on NPM for use with a module bund
+  formatForResponse({ firstName, lastName }: IPet): any {
+    return { firstName, lastName };
+  }
+  formatFromRequest(body: any): any {
+    const { firstName, lastName } = body;
+    return { firstName, lastName };
+  }
+}
+```
+
+Now I can support the new data structure without breaking the old one, and encourage consumers of the old one to switch. I can then remove very little code, as this adapting happened at the right layer insulating most of my code from it.
+
+> Of note, the second parameter to the format methods is `req` in the rare case you need something raw from the req, eg a query param that is doing column filtering.
+
+## What if I need a different handler based on the media type?
+TODO
+
+## What if I need to lock down most of my endpoints?
+So far, we haven't done any authorization, lets lock it down.
+
+If we remove allow anonymous, we will start getting a 401
+```ts
+new RouterMetaBuilder()
+  .path('/pets/')
+  .mediaType(this.petsMediaType)
+  .get(() => {
+    return this.pets;
+  }),
+```
+
+```sh
+curl -H 'accept:application/pets+json'  http://localhost:4000/pets/
+{"status":401,"message":"Authorization is required"}
+```
+
+Security is going to be different for every application, so we provide a hook for you to bring your own security.
+
+
+```ts
+
+import 'reflect-metadata'; // Required by aurelia-dependency-injection
+
+import { ISecurityContext, ISecurityContextProvider, RouteProvider, SecurityContextProvider } from '@symbiotic/express-router-adapter';
+import { Container } from 'aurelia-dependency-injection';
+import { ApplicationRouteProvider } from './ApplicationRouteProvider';
+
+class ApplicationSecurityContext implements ISecurityContext {
+  constructor(public principal: string) { }
+  toLogSafeString(): string {
+    return this.principal;
+  }
+}
+// tslint:disable-next-line: max-classes-per-file
+class ApplicationSecurityContextProvider implements ISecurityContextProvider {
+  async getSecurityContext({ req }: any): Promise<ApplicationSecurityContext> {
+    console.log('in here?');
+    const authHeader = req.headers.authorization;
+    // this is NOT real security
+    return new ApplicationSecurityContext(authHeader);
+  }
+
+}
+
+export const getContainer = async (): Promise<Container> => {
+  const container = new Container();
+
+  container.registerAlias(ApplicationRouteProvider, RouteProvider);
+  container.registerAlias(ApplicationSecurityContextProvider, SecurityContextProvider);
+
+  return container;
+};
+```
+
+Now we get the following
+```sh
+curl -H 'authorization:jon' -H 'accept:application/pets+json'  http://localhost:4000/pets/
+[{"firstName":"honey","lastName":"hoguet"},{"firstName":"poseidon","lastName":"hoguet"}]
+```
+
+Which of course isn't secure! But it demonstrates the plumbing.
+
+> Of note: you must have a principal on your security context to not be considered anonymous. You will still get a 401 without a principal property. This should be reconsidered in the next version.
```

Comment:
`This should be reconsidered in the next version ` The readme is probably not the right place to mention this?